src/components/application_manager/test/hmi_capabilities_test.cc

```diff
@@ -286,84 +283,78 @@ class HMICapabilitiesTest : public ::testing::Test {
   HMICapabilitiesTest()
       : last_state_wrapper_(std::make_shared<resumption::LastStateWrapperImpl>(
             std::make_shared<resumption::LastStateImpl>(kAppStorageFolder,
-                                                        kAppInfoStorage)))
-      , file_name_(kHmiCapabilitiesDefaultFile)
-      , file_cache_name_(kHmiCapabilitiesCacheDefaultFile) {}
-  virtual void SetUp() OVERRIDE {
+                                                        kAppInfoStorage))) {
     if (languages_map.empty()) {
       InitLanguageCStringToEnumMap(languages_map);
     }
     if (light_names_map.empty()) {
       InitLightNameCStringToEnumMap(light_names_map);
     }
+  }
 
-    ON_CALL(mock_app_mngr_, IsSOStructValid(_, _)).WillByDefault(Return(true));
-
-    EXPECT_CALL(mock_app_mngr_, event_dispatcher())
-        .WillOnce(ReturnRef(mock_event_dispatcher));
-    EXPECT_CALL(mock_app_mngr_, get_settings())
-        .WillRepeatedly(ReturnRef(mock_application_manager_settings_));
+  void SetUp() OVERRIDE {
+    ON_CALL(mock_app_mngr_, event_dispatcher())
+        .WillByDefault(ReturnRef(mock_event_dispatcher_));
+    ON_CALL(mock_app_mngr_, get_settings())
+        .WillByDefault(ReturnRef(mock_application_manager_settings_));
     ON_CALL(mock_application_manager_settings_, hmi_capabilities_file_name())
-        .WillByDefault(ReturnRef(file_name_));
+        .WillByDefault(ReturnRef(kHmiCapabilitiesDefaultFile));
     ON_CALL(mock_application_manager_settings_,
             hmi_capabilities_cache_file_name())
-        .WillByDefault(ReturnRef(file_cache_name_));
-    EXPECT_CALL(mock_event_dispatcher, add_observer(_, _, _));
-    EXPECT_CALL(mock_event_dispatcher, remove_observer(_));
-    ON_CALL(mock_application_manager_settings_, launch_hmi())
-        .WillByDefault(Return(false));
-    hmi_capabilities_test =
+        .WillByDefault(ReturnRef(kHmiCapabilitiesCacheFile));
+
+    hmi_capabilities_ =
         std::make_shared<HMICapabilitiesForTesting>(mock_app_mngr_);
-    hmi_capabilities_test->Init(last_state_wrapper_);
   }
 
   void TearDown() OVERRIDE {
-    DeleteFileIfExists(kHmiCapabilitiesCacheDefaultFile);
-    hmi_capabilities_test.reset();
-  }
-  static void TearDownTestCase() {
+    DeleteFileIfExists(kHmiCapabilitiesCacheFile);
     DeleteFileIfExists(kAppInfoDataFile);
+    hmi_capabilities_.reset();
   }
 
-  static void DeleteFileIfExists(const std::string& file_path) {
+  void CreateFile(const std::string& file_name) {
+    file_system::CreateFile(file_name);
+    ASSERT_TRUE(file_system::FileExists(file_name));
+  }
+
+  void DeleteFileIfExists(const std::string& file_path) {
     if (file_system::FileExists(file_path)) {
       EXPECT_TRUE(::file_system::DeleteFile(file_path));
     }
   }
 
-  void SetCooperating();
+  std::shared_ptr<HMICapabilitiesForTesting> hmi_capabilities_;
```

Comment:
@ydementieiev 
- why do you have header file `#include "application_manager/hmi_capabilities.h"` if you use HMICapabilitiesForTesting ? Do you use somewhere in the test real HMI Capabilities? 
- Why do we need at all this class `HMICapabilitiesForTesting`? Can't we test real HMICapabilities class?

Reply:
@AByzhynar I double checked,and deleted HMICapabilitiesForTesting, now we using real HMICapabilities class. Thank you